intro/integration/js.md

```diff
@@ -23,44 +29,141 @@ import skygear from 'skygear/react-native'
 ```
 
 
-# Using older version
+# Configurating
+
+You can use the following code snippet to configure the SDK to connect to server.
+
+```
+skygear.config({
+  'endPoint': 'https://<your-app-name>.skygeario.com/', // trailing slash is required
+  'apiKey': '<your-api-key>',
+}).then(() => {
+  console.log('skygear container is now ready for making API calls.');
+}, (error) => {
+  console.error(error);
+});
+```
+
+## Angular and Ionic
+
+A project using Angular (including Ionic) should follow Angular's pattern of managing data. Skygear is usually used as a global singleton object, the Angular way to manage such data is through [Service](https://angular.io/docs/ts/latest/tutorial/toh-pt4.html).
 
-For versions before 0.24, some manual works are needed for integrating frameworks building with webpack, popular examples are Ionic and Angular.
+First, you need to create a `SkygearService` that hold an refernce of the singleton skygear object. An empty service looks like this, save this as `skygear.service.ts`.
+
+```
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SkygearService {
+}
+```
+
+Then you have to add a method to the service for configurating and returing the skygear instance.
+
+```
+  isConfigurated = false;
+  getSkygear() {
+    if (this.isConfigurated) {
+      // Return the instance directly if it is ready
+      return Promise.resolve(skygear);
+    }
+    let promise = skygear.config({
+      'endPoint': 'https://<your-app-name>.skygeario.com/', // trailing slash is required
+      'apiKey': '<your-api-key>',
+    });
+    promise.then(()=> this.isConfigurated = true);
+    return promise;
+  }
+```
```

Comment:
I don't think telling them to use `ngOnInit` means teaching them about life cycle. I think it is like, "You don't need to know what does `ngOnInit` mean, I am telling you this is the right place to put the code."

Also, this `getSkygear()` function is inside `SkygearService`, not the component. A service does not know when is the function called and I think we cannot make any assumption in the service. The safest way would be returning whatever `skygear.config()` returns, which is a promise.

And, there is no guarantee that the config promise will finish before the UI loads (otherwise `config` should not return a promise at all). We know about the internal implementation of `config` so we are confident to assume "by the time the button is clicked, the skygear object should have finished config", but this assumption is not clear for user. If we are not using a promise interface, we have to explain something like "Since `skygear.config` is really fast, we here assume it is already done"